extensions/wikia/Flags/tasks/FlagsLogTask.php

```diff
@@ -8,18 +8,22 @@ class FlagsLogTask extends BaseTask {
 	/**
 	 * Task for adding logs about changed flags to Special:Log and Special:RecentChanges
 	 * It adds one log per changed flag
+	 *
+	 * Task need to be run in wikia context (where flags were changed) to store in local logging table in DB -
+	 * invoke ->wikiId( ... ) method before queuing task.
+	 *
 	 * @param array $flags list of flags changed, each item of that list is an array with flag fields as items
-	 * @param int $wikiId ID of wiki where the article is
 	 * @param int $pageId ID of article where flags were changed
 	 * @param string $actionType Type of action performed on flag represented by constants in Flags\Models\Flag class
 	 */
-	public function logFlagChange( array $flags, $wikiId, $pageId, $actionType ) {
+	public function logFlagChange( array $flags, $pageId, $actionType ) {
+		global $wgCityId;
```

Comment:
You can overwrite the `wiki_id` variable when making an API request so I think we should be consistent here and log othe event on a wikia that has been specified in the request (and I consider not using a custom `wiki_id` parameter specifying as well :) ).